launcher/src/main/java/org/wildfly/core/launcher/StandaloneCommandBuilder.java

```diff
@@ -137,8 +138,17 @@ public StandaloneCommandBuilder addJavaOption(final String jvmArg) {
      */
     public StandaloneCommandBuilder addJavaOptions(final String... javaOpts) {
         if (javaOpts != null) {
-            for (String javaOpt : javaOpts) {
-                addJavaOption(javaOpt);
+            for (int i = 0; i < javaOpts.length; i++) {
+                if (Arguments.isModuleSystemArgument(javaOpts[i])) {
+                    // handle java module system options
+                    if (i < javaOpts.length - 1) {
+                        this.javaOpts.add(Arguments.JavaModuleSystemArgument.of(javaOpts[i], javaOpts[++i]));
+                    } else {
+                        // ignore corrupted module system argument
```

Comment:
Same as above, we should just add the arguments and allow the command to fail when executed.